src/Dependencies/Collections/SegmentedArray`1.cs

```diff
@@ -386,10 +386,10 @@ public Enumerator(SegmentedArray<T> array)
                 _current = default!;
             }
 
-            public T Current => _current;
+            public readonly T Current => _current;
             object? IEnumerator.Current => Current;
```

Comment:
❔ Why is this one not marked `readonly`, but the one below is?

Reply:
so the system is not multi-pass.  so first pass mae `T Current` readonly.  Another pass would then let followup members become readonly.  This is similar to how 'make static' and otehr similar features work.